### PR TITLE
Study nurse menu feature

### PR DESCRIPTION
**Pull request by mustiwebp**
What the title says.

### CODE REVIEW

**Comment by S1rDan** [comment]
When I try to edit a patient I'm getting a LateInitializationError: Field '_weightBMI@99384301' has not been initialized.

File: lib/screens/study_nurse_screen/menu_screen/patients_list.dart [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                );
+              }
+
+              return ListView.builder(
+                itemCount: statePatientData.objectsList.length,
+                itemBuilder: (BuildContext context, int index) {
+                  for (int j = 0; j < statePatient.objectsList.length; j++) {
+                    if ((statePatient.objectsList[j] as Patient).objectId ==
+                        (statePatientData.objectsList[index] as PatientData)
+                            .patientObjectId) {
```
Comment (S1rDan) [inline lib/screens/study_nurse_screen/menu_screen/patients_list.dart:76]: This process is too expensive and gets even more so if the amount of data grows.

It should be handled differently:
Either you build a new query and let the database handle that issue by applying filters etc.
or you allow all Patient objects and let the user fill the missing information when editing the entry.

File: lib/widgets/picos_list_card.dart
Comment (S1rDan) [inline lib/widgets/picos_list_card.dart:1]: There are more changes to be made.
If you look closely the singular button is not symmetric.

**Review by S1rDan — CHANGES_REQUESTED** [review]

File: lib/screens/study_nurse_screen/menu_screen/patients_list.dart [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+              if (statePatient.status == ObjectsListStatus.failure ||
+                  statePatientData.status == ObjectsListStatus.failure) {
+                return const Center(
+                  child: Text('Error'),
+                );
+              }
+
+              return ListView.builder(
+                itemCount: statePatientData.objectsList.length,
```
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/screens/study_nurse_screen/menu_screen/patients_list.dart:71]: Well, I took the minimum of both.

File: lib/screens/study_nurse_screen/menu_screen/add_patient_screen.dart
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/screens/study_nurse_screen/menu_screen/add_patient_screen.dart:1]: I think it's easier than your way.

File: lib/screens/study_nurse_screen/menu_screen/patients_list.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (S1rDan) [reply lib/screens/study_nurse_screen/menu_screen/patients_list.dart]: Thing is, is it granted that the PatientData count ist always the same as the Patients count to be shown?

**Pull request closed without merging**